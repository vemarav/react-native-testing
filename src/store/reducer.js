import {createSlice} from '@reduxjs/toolkit';

const reducer = createSlice({
  name: 'reducer',
  initialState: '',
  reducers: {
    updateText: (state, action) => {
      return action.payload;
    },
  },
});

export const {updateText} = reducer.actions;
export default reducer.reducer;
