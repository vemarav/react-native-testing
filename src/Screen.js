import React, {useEffect, useState} from 'react';
import {View, Text, Button} from 'react-native';

import {updateText} from './store/reducer';
import {useDispatch, useSelector} from 'react-redux';

const Screen = () => {
  const text = useSelector(state => state.reducer);
  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(updateText('mounted'));
  }, []);

  return (
    <View style={{backgroundColor: 'black'}} testId="container">
      <Text testID="text">{text}</Text>
      <Button
        testID="button"
        onPress={() => dispatch(updateText('button pressed'))}
        title={'Update text'}
      />
    </View>
  );
};

export default Screen;
