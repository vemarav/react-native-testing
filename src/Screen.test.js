import React from 'react';
import {create, act} from 'react-test-renderer';
import Screen from './Screen';

import store from './store';
import {Provider} from 'react-redux';

const tree = create(
  <Provider store={store}>
    <Screen />
  </Provider>,
);

jest.mock('react-native/Libraries/Animated/NativeAnimatedHelper');
jest.runAllTimers();

it('snapshot', () => {
  expect(tree).toMatchSnapshot();
});

it('press button', () => {
  expect(store.getState().reducer).toEqual('mounted');
  const button = tree.root.findByProps({testID: 'button'}).props;
  act(() => button.onPress());

  const text = tree.root.findByProps({testID: 'text'}).props;
  expect(text.children).toEqual('button pressed');
});

it('text updated properly', () => {
  expect(store.getState().reducer).toEqual('button pressed');
});
