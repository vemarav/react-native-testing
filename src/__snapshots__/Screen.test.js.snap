// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<View
  style={
    Object {
      "backgroundColor": "black",
    }
  }
  testId="container"
>
  <Text
    testID="text"
  >
    mounted
  </Text>
  <View
    accessibilityRole="button"
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID="button"
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "margin": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Update text
      </Text>
    </View>
  </View>
</View>
`;
